x-pack/legacy/plugins/apm/public/components/shared/TransactionActionMenu/sections.ts

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { AppMountContext } from 'kibana/public';
+import { Location } from 'history';
+import url from 'url';
+import { pick } from 'lodash';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { getInfraHref } from '../Links/InfraLink';
+import { getDiscoverQuery } from '../Links/DiscoverLinks/DiscoverTransactionLink';
+import { getDiscoveryHref } from '../Links/DiscoverLinks/DiscoverLink';
+import { fromQuery } from '../Links/url_helpers';
+import { IUrlParams } from '../../../context/UrlParamsContext/types';
+
+function getInfraMetricsQuery(transaction: Transaction) {
+  const plus5 = new Date(transaction['@timestamp']);
+  const minus5 = new Date(plus5.getTime());
+
+  plus5.setMinutes(plus5.getMinutes() + 5);
+  minus5.setMinutes(minus5.getMinutes() - 5);
```

Comment:
i wonder if you can leverage `@elastic/datemath` here:

```
const plus5 = datemath.parse(`${transaction['@timestamp']} + 5m`)
const minus5 = datemath.parse(`${transaction['@timestamp']} - 5m`)
```

Reply:
I don't think `datemath` support such syntax @ogupte

Reply:
I remember stumbling upon this and thinking about changing it. What do you think about this approach:

```ts
function getInfraMetricsQuery(transaction: Transaction) {
  const timestamp = new Date(transaction['@timestamp']).getTime();
  const fiveMinutes = 1000 * 60 * 5;

  return {
    from: timestamp - fiveMinutes,
    to: timestamp + fiveMinutes
  };
}
```

Reply:
Instead of:
```js
  const fiveMinutes = 1000 * 60 * 5;
```

You can do:
```js
  const fiveMinutes = moment.duration(5, 'minutes').asMilliseconds();
```

But I don't thinks it's worth it tbh.